src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -321,6 +328,56 @@ export const handleGetPublicForm: ControllerHandler<
           }
           return res.json({ form: publicForm, isIntranetUser })
         })
+    case FormAuthType.SGID_MyInfo: {
+      const accessTokenCookie = req.cookies[SGID_MYINFO_COOKIE_NAME]
+      if (!accessTokenCookie) {
+        return res.json({
+          form: publicForm,
+          isIntranetUser,
+        })
+      }
+      res.clearCookie(SGID_MYINFO_COOKIE_NAME)
+      res.clearCookie(MYINFO_LOGIN_COOKIE_NAME)
+      return SgidService.extractSgidJwtMyInfoPayload(accessTokenCookie)
+        .asyncAndThen((auth) =>
+          SgidService.retrieveUserInfo({ accessToken: auth.accessToken }),
+        )
+        .andThen((userInfo) => {
+          const data = new SGIDMyInfoData(userInfo.data)
+          return MyInfoService.prefillAndSaveMyInfoFields(
+            form._id,
+            data,
+            form.toJSON().form_fields,
+          ).map((prefilledFields) => {
+            return res
+              .cookie(
+                MYINFO_LOGIN_COOKIE_NAME,
```

Comment:
Qn: why use the myinfo_login_cookie instead of sgid_myinfo_cookie?

Reply:
SGID_MYINFO_COOKIE - stores access token for sgID userinfo query.
MYINFO_LOGIN_COOKIE - stores singpass username and is required for MyInfo field's pre-existing functions to work properly.